wren-ui/src/apollo/server/services/askingService.ts

```diff
@@ -844,6 +872,97 @@ export class AskingService implements IAskingService {
     return updatedThreadResponse;
   }
 
+  public async generateThreadResponseChart(
+    threadId: number,
+    threadResponseId: number,
+    configurations: { language: string },
+  ): Promise<ThreadResponse> {
+    const thread = await this.threadRepository.findOneBy({
+      id: threadId,
+    });
+    const threadResponse = await this.threadResponseRepository.findOneBy({
+      id: threadResponseId,
+    });
+
+    if (!thread) {
+      throw new Error(`Thread ${threadId} not found`);
+    }
+
+    if (!threadResponse) {
+      throw new Error(`Thread response ${threadResponseId} not found`);
+    }
+
+    // 1. create a task on AI service to generate the chart
+    const response = await this.wrenAIAdaptor.generateChart({
+      query: threadResponse.question,
+      sql: threadResponse.sql,
+      configurations,
+    });
+
+    // 2. update the thread response with chart detail
+    const updatedThreadResponse = await this.threadResponseRepository.updateOne(
+      threadResponse.id,
+      {
+        chartDetail: {
+          queryId: response.queryId,
+          status: ChartStatus.FETCHING,
+        },
+      },
+    );
+
+    // 3. put the task into background tracker
+    this.chartBackgroundTracker.addTask(updatedThreadResponse);
+
+    return updatedThreadResponse;
+  }
+
+  public async adjustThreadResponseChart(
+    threadId: number,
+    threadResponseId: number,
+    input: ChartAdjustmentOption,
+    configurations: { language: string },
+  ): Promise<ThreadResponse> {
+    const thread = await this.threadRepository.findOneBy({
+      id: threadId,
+    });
+    const threadResponse = await this.threadResponseRepository.findOneBy({
+      id: threadResponseId,
+    });
+
+    if (!thread) {
```

Comment:
I think checking whether the threadResponse exists is enough here.

Reply:
Its removed by following the text-based answer api PR
pls check 59b367c7